src/main/java/duke/commands/DeleteTask.java

```diff
@@ -0,0 +1,46 @@
+package duke.commands;
+
+import duke.ui.Ui;
+import duke.tasks.Task;
+import duke.tasks.TaskList;
+import duke.storage.Storage;
+import duke.parsers.CommandParser;
+import duke.exceptions.DukeException;
+
+/**
+ * Deletes a Task from the TaskList.
+ */
+class DeleteTask extends Command {
+
+    public DeleteTask(CommandParser commandParser) {
+        super(commandParser);
+    }
+
+    public void execute(String arg, TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        int taskNo = getTaskNo(arg);
+
+        Task removed = removeTask(tasks, taskNo);
+
+        ui.showReply(String.format("Noted. I've removed this task:\n  %s\nNow you have %d tasks in the list.", removed,
```

Comment:
Our static analysis engine detects that your pull request includes a change that a format string `\n` instead of `%n`. According to [this description](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline), it is a bad practice, since using `\n` in format string may cause cross-platform issues.